src/3rd_party_bundle_add.c

```diff
@@ -94,18 +96,134 @@ static bool parse_options(int argc, char **argv)
 	return true;
 }
 
-static enum swupd_code add_bundle(char *bundle)
+static enum swupd_code create_wrapper_script(char *filename)
+{
+	enum swupd_code ret_code = 0;
+	int fd;
+	FILE *fp = NULL;
+	char *bin_directory = NULL;
+	char *script = NULL;
+	char *binary = NULL;
+	char *env_path = NULL;
+	char *env_ld_path = NULL;
+	char *third_party_bin_path = NULL;
+	char *third_party_ld_path = NULL;
+	mode_t mode = 0755;
+
+	bin_directory = third_party_get_bin_dir();
+	script = third_party_get_binary_path(sys_basename(filename));
+	binary = sys_path_join(globals.path_prefix, filename);
+
+	if (!sys_filelink_is_executable(binary)) {
+		goto close_and_exit;
```

Comment:
It might be worthwhile to have some way to notify the user that the mix bundle they installed isn't configured properly so they can know something about the quality of the mix they are using.

Reply:
Hmm this comment applied to part of the code was really part of PR #1303, which has already merged. I'll address this comment in a separate PR.